flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/util/serialization/KeyedDeserializationSchema.java

```diff
@@ -32,6 +34,49 @@
  */
 @PublicEvolving
 public interface KeyedDeserializationSchema<T> extends Serializable, ResultTypeQueryable<T> {
+	/**
+	 * Kafka record to be deserialized.
+	 * Record consists of key,value pair, topic name, partition offset, headers and a timestamp (if available)
+	 */
+	interface Record {
```

Comment:
I am wondering if we should just use ConsumerRecord from kafka directly. this `KeyedDeserializationSchema` is for kafka consumer only. there isn't much benefit of defining another Flink interface.

just to give a counter point. I am also interested in the `timestamp` fields. now we need to expand this interface.

Reply:
I already answered before: we had to keep KeyedDeserializationSchema.Record because KeyedDeserializationSchema is used by legacy connectors, e.g. Kafka 0.8 doesn't have _ConsumerRecord_ at all
Regarding timestamp - yes, we will need to expand interface, and I planned address it in separate PR on top of this one.

Reply:
@alexeyt820 as pointed out by @azagrebin, kafka 0.8 seems to have `ConsumerRecord` interface
https://archive.apache.org/dist/kafka/0.8.2-beta/java-doc/org/apache/kafka/clients/consumer/ConsumerRecord.html

As Flink is moving to the direction of just one modern kafka connector, I am also wondering if it is ok to drop kafka 0.8 connector?

Reply:
We can add a new deserialize method to `KeyedDeserializationSchema` interface with a default implementation that just forwards to the other deserialize method (mark as deprecated).

{code}
T deserialize(ConsumerRecord consumerRecord) throws IOException {
    return deserialize(messageKey, message, topic, partition, offset);
}
{code}

Reply:
Done in 2adbacb